caddyhttp/proxy/reverseproxy.go

```diff
@@ -254,6 +254,10 @@ func (rp *ReverseProxy) copyResponse(dst io.Writer, src io.Reader) {
 
 func copyHeader(dst, src http.Header) {
 	for k, vv := range src {
+		// some predefined headers (e.g. MIME plugin), skip.
+		if _, ok := dst[k]; ok {
+			continue
+		}
```

Comment:
So basically, this says that if the destination (whether upstream or downstream) already has the header field specified, we do not add it again. I can see how this fixes Content-Type problems, but what if we want to follow the Server header through all the proxy points?

I wonder if this check should be for certain fields only... what do you think?